platforms/jvm/ear/src/main/java/org/gradle/plugins/ear/EarPlugin.java

```diff
@@ -109,9 +110,7 @@ private void setupEarTask(final Project project, PluginContainer plugins) {
                 component.getMainFeature().getSourceSet().getResources().srcDir(task.getAppDirectory());
             });
 
-            DeploymentDescriptor deploymentDescriptor = objectFactory.newInstance(DefaultDeploymentDescriptor.class);
-            deploymentDescriptor.readFrom("META-INF/application.xml");
-            deploymentDescriptor.readFrom("src/main/application/META-INF/" + deploymentDescriptor.getFileName());
+            DeploymentDescriptor deploymentDescriptor = createDeploymentDescriptor();
             if (deploymentDescriptor != null) {
```

Comment:
:x: `deploymentDescriptor` cannot be null if created by the previous statement

Reply:
agree :)